pkg/util/profiling/runtime.go

```diff
@@ -0,0 +1,41 @@
+package profiling
+
+import (
+	"runtime"
+)
+
+var (
+	// Go runtime does not provide a method to read current value of block profiling rate,
+	// but do need it in a few places: to restore state after collecting flare profile, and
+	// to pass to continuous profiler so it doesn't overwrite it with built-in default.
+	blockProfileRate int
+)
+
+// SetBlockProfileRate sets goroutine blocking profile rate.
+func SetBlockProfileRate(rate int) {
+	mu.Lock()
+	defer mu.Unlock()
+	blockProfileRate = rate
+	runtime.SetBlockProfileRate(rate)
+}
+
+// GetBlockProfileRate returns the current rate of goroutine blocking profile.
+func GetBlockProfileRate() int {
+	mu.RLock()
+	defer mu.RUnlock()
+	return blockProfileRate
+}
+
+// SetMutexProfileFraction sets fraction of mutexes that generate profiling data.
+func SetMutexProfileFraction(fraction int) {
+	mu.Lock()
+	defer mu.Unlock()
+	runtime.SetMutexProfileFraction(fraction)
```

Comment:
Does this call only set a profiling option on the Go runtime, or does it also make the Go runtime start profiling?

(same question about `runtime.SetBlockProfileRate`)

Reply:
Yes, setting these to non-zero values makes go runtime to start collecting profiling data.

Reply:
ok, is there any risk to leaving them on "forever"? (does it use more and more memory over time for example?). If so, I think these runtime settings should be hidden (and worth a comment in the code explaining why they're hidden)

Reply:
These features do allocate a small amount per-stack frame sampled. I would expect allocations to taper off with time, as the profiler accumulates location data; but there is risk that number of possible stack frames to be too large. This would cause not only increased memory usage, but also additional cpu cost (profiling data is stored in a simple fixed-size linked-list hashmap with a fairly simple hash function). Since this is a concern, I will make these settings hidden.